docs/03-usage.md

```diff
@@ -76,11 +89,59 @@ npx snowpack --help
+
+```json
+{
+  "webDependencies": [
+    "preact",
+  ],
+  "options": {
+    "optimize": false
+  }
+}
+```
+
+#### Using snowpack.config.js
+
+To generate parts of your configuration with Node.js, you may use a `snowpack.config.js` instead:
+
+```js
+module.exports = {
+  webDependencies: [...myWebDependenciesGeneratorFunction()],
+  options: {
+    optimize: process.env.NODE_ENV === "production",
+    strict: true
+  }
+};
 ```
 
+#### All Options
 
+| Key               | Type       | Default | Description                                                                                                                                                                                                                                                                                                                                                                                                                            |
+| ----------------- | ---------- | ------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `webDependencies` | `string[]` | `[]`    | (Recommended) Set exactly which packages to install with Snowpack. Without this, Snowpack will just try to install every package in your "dependencies" config. That behavior is great for getting started but it won't warn you if an expected package fails to install.                                                                                                                                                              |
+| `options.*`       | `object`   | `{}`    | CLI options (camelcased, so `source-map` becomes `"sourceMap"`). If any settings here conflict with a CLI flag, the CLI flag takes priority.                                                                                                                                                                                                                                                                                           |  |
+| `namedExports`    | `object`   |         | (Optional) If needed, you can explicitly define named exports for any dependency. You should only use this if you're getting `"'X' is not exported by Y"` errors without it. See [rollup-plugin-commonjs](https://github.com/rollup/rollup-plugin-commonjs#usage) for more info.                                                                                                                                                       |
```

Comment:
while we're at it, this might be a good time to put  `namedExports` under a "rollupOptions" namespace.

"dedupe" is technically a rollup option but its such a high-level concept that i think it works better as a top-level option.

Reply:
I’m +1 but maybe let’s hold off on that. Because the way this works, changing the order also changes the `package.json` structure, so it’d either be a breaking change, or we’d have to duplicate the config and figure out how to resolve conflicts. We only get to add `options` freely because all the CLI commands are new to the `package.json` config.

Reply:
Oh right, of course. Yea lets hold off then, especially if we're only talking about the one option for now. Thanks for the sanity check :)